File: javascript/ExpirableProduct.js
// javascript/ExpirableProduct.js
import { Product } from './Product.js';

export class ExpirableProduct extends Product {
    constructor(name, price, quantity, expiryDate) {
        super(name, price, quantity);
        if (this.constructor === ExpirableProduct) {
            throw new Error("Abstract classes can't be instantiated.");
        }
        this.expiryDate = expiryDate;
    }

    getExpiryDate() {
        return this.expiryDate;
    }

    isExpired() {
        return new Date() > this.expiryDate;
    }
}

File: python/expirable_product.py
from datetime import date
from product import Product

class ExpirableProduct(Product):
    def __init__(self, name, price, quantity, expiry_date):
        super().__init__(name, price, quantity)
        self.expiry_date = expiry_date
    
    def get_expiry_date(self):
        return self.expiry_date
    
    def is_expired(self):
        return date.today() > self.expiry_date
